index.js

```diff
@@ -14,11 +14,39 @@ module.exports = function (opts) {
   var ip = opts.ip || opts.host || (type === 'udp4' ? '224.0.0.251' : null)
   var me = {address: ip, port: port}
   var destroyed = false
+  var interfaces = opts.interface ? (Array.isArray(opts.interface) ? opts.interface : [opts.interface]) : []
+  var sendSockets = []
 
-  if (type === 'udp6' && (!ip || !opts.interface)) {
+  if (type === 'udp6' && (!ip || !interfaces.length)) {
     throw new Error('For IPv6 multicast you must specify `ip` and `interface`')
   }
 
+  // create sockets to send messages
+  for (var si = 0; si <= interfaces.length; si++) {
+    sendSockets[si] = dgram.createSocket({
+      type: type,
+      reuseAddr: opts.reuseAddr !== false,
+      toString: function () {
+        return type
+      }
+    })
+
+    sendSockets[si].on('listening', function () {
+      if (opts.multicast !== false) {
+        this.setMulticastTTL(opts.ttl || 255)
+        this.setMulticastLoopback(opts.loopback !== false)
```

Comment:
This is not enough to send packets on this interface. For example, Linux will select default interface.

To achieve sending via proper interface the call `this.setMulticastInterface(iterfaceAddress)` is required. Because this is a callback, I suggest convert entire loop body to immediately invoked function with interface address as a parameter.